docs/appendix-A.rst

```diff
@@ -0,0 +1,1783 @@
+   :language: c
+   :start-after: start callback1
+   :end-before: end callback1
+
+YAML:
+
+.. code-block:: yaml
+
+    - decl: int callback1(int in, int (*incr)(int));
+
+The C wrapper:
+
+.. literalinclude:: ../regression/reference/tutorial/wrapTutorial.cpp
+   :language: c
+   :start-after: start TUT_callback1
+   :end-before: end TUT_callback1
+
+Creates the abstract interface:
+
+.. literalinclude:: ../regression/reference/tutorial/wrapftutorial.f
+   :language: fortran
+   :start-after: start abstract callback1_incr
+   :end-before: end abstract callback1_incr
+   :dedent: 8
+
+Calls C via the interface:
+
+.. literalinclude:: ../regression/reference/tutorial/wrapftutorial.f
+   :language: fortran
+   :start-after: start callback1
+   :end-before: end callback1
+   :dedent: 8
+
+Fortran usage:
+
+.. code-block:: fortran
+
+    module worker
+      use iso_c_binding
+    contains
+      subroutine userincr(i) bind(C)
+        integer(C_INT), value :: i
+        ! do work of callback
+      end subroutine user
+
+      subroutine work
+        call callback1(1, userincr)
+      end subrouine work
+    end module worker
+
+
+.. ############################################################
+
+.. _example_callback1c:
+
+callback1c
+^^^^^^^^^^
+
+C library function:
+
+.. literalinclude:: ../regression/run/clibrary/clibrary.c
+   :language: c
+   :start-after: start callback1
+   :end-before: end callback1
+
+YAML:
+
+.. code-block:: yaml
+
+    - decl: int callback1(int type, void (*incr)()+external)
+
+Creates the abstract interface:
+
+.. literalinclude:: ../regression/reference/clibrary/wrapfclibrary.f
+   :language: fortran
+   :start-after: start abstract callback1_incr
+   :end-before: end abstract callback1_incr
+   :dedent: 8
+
+Calls C via the interface:
+
+.. literalinclude:: ../regression/reference/clibrary/wrapfclibrary.f
+   :language: fortran
+   :start-after: start c_callback1
+   :end-before: end c_callback1
+   :dedent: 8
+
+.. XXX why is C_PTR used here ^
+
+The Fortran wrapper.
+By using ``external`` no abstract interface is used:
+
+.. literalinclude:: ../regression/reference/clibrary/wrapfclibrary.f
+   :language: fortran
+   :start-after: start callback1
+   :end-before: end callback1
+   :dedent: 4
+
+Fortran usage:
+
+.. code-block:: fortran
+
+    module worker
+      use iso_c_binding
+    contains
+      subroutine userincr_int(i) bind(C)
+        integer(C_INT), value :: i
+        ! do work of callback
+      end subroutine user_int
+
+      subroutine userincr_double(i) bind(C)
+        real(C_DOUBLE), value :: i
+        ! do work of callback
+      end subroutine user_int
+
+      subroutine work
+        call callback1c(1, userincr_int)
+        call callback1c(1, userincr_double)
+      end subrouine work
+    end module worker
+
+Struct
+------
+
+.. ############################################################
+
+.. _example_passStruct1:
+
+passStruct1
+^^^^^^^^^^^
+
+C library function:
```

Comment:
It would be helpful here to include how to tell shroud about the `Cstruct`

Reply:
I need to fill out the struct/class stuff more.  I'll do that in another pull request.